### PR TITLE
Fix MaterialsProjectCompatibility run type handling for GGA+U

### LINKED ISSUE #3506: Vasprun return invalid run type PBE
Hi, I would like to apply the legacy corrections with MaterialsProjectCompatibility. However, an invalid run-type was returned.
```
from pymatgen.io.vasp import Vasprun
from pymatgen.entries.compatibility import MaterialsProjectCompatibility, AnionCorrection
vasprun1 = Vasprun("data/mp_1595_vasprun.xml")
entry1 = vasprun1.get_computed_entry(inc_structure=True)
compatibility = MaterialsProjectCompatibility()
entry_ = compatibility.get_adjustments(entry1)
```
Error
```
CompatibilityError: Entry vasprun-2023-12-08 10:44:39.983016 has invalid run type PBE. Discarding.
```
[mp_1595_vasprun.xml.zip](https://github.com/materialsproject/pymatgen/files/13608078/mp_1595_vasprun.xml.zip)

### PR DESCRIPTION
**Pull request by rkingsbury**
Fixes #3506

See discussion in #3534 which settled on moving the run type check and `raise` statement to `get_adjustments` itself and remove the other statements, as illustrated [here](https://github.com/materialsproject/pymatgen/pull/3540/files) as the cleanest fix.

The logic is the following:
- when `process_entries` is called, it calls `get_adjustments` for every entry provided
- Any entry which raises a `CompatibilityError` is excluded from the returned list.
- In the specific case of `MaterialsProjectCompatibility`, `get_adjustments` calls `get_corrections_dict`, which loop through the `get_correction` method of all 4 of the underlying `Correction` classes (we eliminated these in the newer scheme for simplicity)
- It just so happens that the `raise` statement that occurs in `UCorrection` (which processes any entry sent through `get_Adjustments`) serves the purpose of excluding incompatible entries. 
- Strictly speaking, this `raise` statement could/should occur in `get_adjustments` itself, because that is really the only place that the run_type check needs to happen (an entry which is incompatible with one of the corrections is incompatible with all of them, for this purpose)
- The `return ufloat(0,0)` statements in the `Correction` classes do not serve an important purpose as far as I can tell.

### CODE REVIEW

**Comment by rkingsbury** [comment]
@materialsproject/pymatgen-maintainers is something amiss with the CI tests? They are all failing with

```
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-h4vu0vp5/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-h4vu0vp5/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-h4vu0vp5/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-h4vu0vp5/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 41, in <module>
      ModuleNotFoundError: No module named 'packaging'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

File: pymatgen/entries/compatibility.py
Comment (rkingsbury) [inline pymatgen/entries/compatibility.py:650]: Actually, since this method is in the base `CorrectionsList` class and not specifically in `MaterialsProjectCompatibility`, I need to add a kwarg to specify what the valid run types are in case there are other compatibility classes derived from this one. Will do that next week.

**Comment by DanielYang59** [comment]
It appears to me the GitHub workflow might be missing something, thus every commit check seems to fail because of the same error: 

```
 × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-d4g7fty8/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-d4g7fty8/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-d4g7fty8/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-d4g7fty8/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 41, in <module>
      ModuleNotFoundError: No module named 'packaging'
      [end of output]

```

I was about to open an issue but maybe I could help fix the issue in workflow next week.

**Comment by rkingsbury** [comment]
Ready to merge, unless other comments / concerns from @materialsproject/pymatgen-maintainers

**Review by janosh — APPROVED** [review]
Thanks @rkingsbury and @DanielYang59! Nice work collaborating on this fix.

**Comment by DanielYang59** [comment]
Good work @rkingsbury ! Learned a lot from my side. And thanks for reviewing @janosh

**Pull request merged**
